client/state/themes/upload-theme/reducer.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	THEME_UPLOAD_START,
+	THEME_UPLOAD_SUCCESS,
+	THEME_UPLOAD_FAILURE,
+	THEME_UPLOAD_CLEAR,
+	THEME_UPLOAD_PROGRESS,
+} from 'state/action-types';
+
+export const uploadedTheme = createReducer( {}, {
+	[ THEME_UPLOAD_SUCCESS ]: ( state, { siteId, theme } ) => ( {
+		...state,
+		[ siteId ]: theme,
```

Comment:
If this is a theme object, I wonder -- should we only store the theme ID here, and use our single-source-of-truth reducer (`queries`) to store the object? Just so we don't duplicate data/sources-of-truth again...